tools/mask_gen.py

```diff
@@ -0,0 +1,93 @@
+import numpy as np
+import cv2
+import argparse
+
+class segment_drawer(object):
+
+    def __init__(self, image_filename=None, output_filename=None):
+        # Read in the image and start the drawing
+        image = cv2.imread(image_filename)
+        self.mask = np.zeros((image.shape[0],image.shape[1],3), dtype='uint8')
+        self.brush_size = 30
+
+        cv2.namedWindow('draw', flags=cv2.WINDOW_NORMAL)
+        cv2.imshow('draw',image)
+        cv2.resizeWindow('draw', 500, 500)
+
+        cv2.namedWindow('mask', flags=cv2.WINDOW_NORMAL)
+        cv2.resizeWindow('mask',500,500)
+
+        # Flag for up or down mouse position
+        self.drawing = False
+        cv2.setMouseCallback('draw', self.select_pixel)
+
+        # Output setting
+        self.save = False
+        self.output_filename = output_filename
+        if output_filename is not None:
+            self.save = True
+
+        while(True):
+            #Generate the image to draw based on image + mask
+            combined_image = cv2.addWeighted(image, 0.6, self.mask, 0.4,0)
+            cv2.imshow('draw', combined_image)
+            cv2.imshow('mask',self.mask)
+
+            #Escape functions
+            k = cv2.waitKey(1) & 0xFF
+            if k == 27:
+                if self.save == True:
+                    while(1):
+                        input_var = input('You have specified a save file, would you like to save before quiting? [y/n]')
+                        if input_var.strip() == 'y':
+                            self.save_mask()
+                            break
+                        elif input_var.strip() == 'n':
+                            break
+                        else:
+                            continue
+                break
+
+            elif k == ord('s'):
+                self.save_mask
```

Comment:
Should this be a function call? i.e. `self.save_mask()` rather than just `self.save_mask`?